apps/api/src/app/portfolio/portfolio.controller.ts

```diff
@@ -230,6 +231,58 @@ export class PortfolioController {
     return { investments };
   }
 
+  @Get('dividends')
+  @UseGuards(AuthGuard('jwt'))
+  public async getDividends(
+    @Headers('impersonation-id') impersonationId: string,
+    @Query('range') dateRange: DateRange = 'max',
+    @Query('groupBy') groupBy?: GroupBy
+  ): Promise<PortfolioDividends> {
+    if (
+      this.configurationService.get('ENABLE_FEATURE_SUBSCRIPTION') &&
+      this.request.user.subscription.type === 'Basic'
+    ) {
+      throw new HttpException(
+        getReasonPhrase(StatusCodes.FORBIDDEN),
+        StatusCodes.FORBIDDEN
+      );
+    }
+
+    let dividends: InvestmentItem[];
+
+    if (groupBy === 'month') {
+      dividends = await this.portfolioService.getInvestments({
```

Comment:
I'm not sure if it's a good idea to mix with investments and all the changes in portfolio calculator.

Another option would be to create a dedicated function `this.portfolioService.getDividends()` which has a query to get all dividends:

```
this.orderService.getOrders({
  filters,
  userCurrency,
  userId,
  types: ['DIVIDEND']
});
```

What do you think?